test/test.js

```diff
@@ -0,0 +1,285 @@
+//#############################################################################
+// require helper node_modules and prepare helper methods and consts
+
+const expect = require('chai').expect;
+const fse = require('fs-extra');
+const path = require('path');
+const compare_version = require('compare-version');
+
+const childproc = require('child_process');
+
+const command = function(command, callback) {
+    const proc = childproc.exec(command, function(error, stdout, stderr) {
+        if (error) {
+            // console.log(error.stack);
+            // console.log('Error Code: '+error.code);
+            // console.log('Reason of error: '+error.signal);
+            console.log('ERROR: ' + stderr);
+        }
+        callback(stdout, stderr);
+    });
+}
+
+const path_to_jsfiddle_downloader = path.normalize(__dirname + '/../bin/jsfiddle-downloader.js');
+if (! fse.pathExistsSync(path_to_jsfiddle_downloader)) {
+  console.log(`ERROR: could not find file under test: ${path_to_jsfiddle_downloader}`);
+  process.exit(1);
+}
+
+const keep_results = process.argv.includes('--keep'); // keep resulting directories and files for inspection
+
+//#############################################################################
+// tests for v0.1.6
+
+describe('jsfiddle-downloader v0.1.6 and above has features as described in README', function() {
+
+    this.timeout(10000);
+
+    const tempdir = 'temp_test_results.0.1.6';
+
+    before(function(done) { // check preconditions
+        const __this = this;
+        command(`${path_to_jsfiddle_downloader} --version`, function(stdout, stderr) {
+          console.log(`  version detected ${stdout}`);
+          if (stderr != '') {
+              __this.skip();
+          } else if (compare_version(stdout, '0.1.6') < 0) {
```

Comment:
I agree with your principle "the tests should be targeting features, not versions".

I grouped tests by versions so that I could verify that (1) features present in '0.1.6' work in both '0.1.6' and '0.1.7', and (2) that features present only in '0.1.7' also work.

If you adopt and merge my changes, then all tests can be grouped together, after a simple test verifying that the version is at least '0.1.7'.